assets/src/editor/events/dates-and-times/editor-date/filter-bar/with-filtered-date-entities.js

```diff
@@ -2,41 +2,27 @@
  * External imports
  */
 import { createHigherOrderComponent } from '@wordpress/compose';
-
-/**
- * Internal imports
- */
-import { getFilteredDateEntitiesList } from './with-date-entities-list-filter-bar';
+import { withSelect } from '@wordpress/data';
 
 const DEFAULT_EMPTY_ARRAY = [];
 
 const withFilteredDateEntities = createHigherOrderComponent(
-	( WrappedComponent ) => ( {
-		eventEntityLoaded,
-		dateEntitiesLoaded,
-		dateEntities,
-		showDates,
-		datesSortedBy,
-		...otherProps
-	} ) => {
-		const isLoaded = eventEntityLoaded && dateEntitiesLoaded;
-		const filteredDateEntities = isLoaded ?
-			getFilteredDateEntitiesList(
-				dateEntities,
-				showDates,
-				datesSortedBy
-			) :
-			DEFAULT_EMPTY_ARRAY;
-		return <WrappedComponent
-			filteredDateEntities={ filteredDateEntities }
-			allDateEntities={ dateEntities }
-			eventEntityLoaded={ eventEntityLoaded }
-			dateEntitiesLoaded={ dateEntitiesLoaded }
-			showDates={ showDates }
-			datesSortedBy={ datesSortedBy }
-			{ ...otherProps }
-		/>;
-	},
+	withSelect(
+		( select ) => {
+			const store = select( 'eventespresso/filter-state' );
+			const { getEntitiesByIds } = select( 'eventespresso/core' );
+			return {
+				filteredDateEntities: getEntitiesByIds(
+					'datetime',
+					store.getFilter(
+						'event-editor-dates-list',
+						'filteredDateIds',
+						DEFAULT_EMPTY_ARRAY
+					)
+				),
+			};
+		}
+	),
```

Comment:
This hoc just makes it easier for consumers to get the current (and subscribe to the) filtered date entities in the store.